src/routes/notifications/v1/notifications.controller.ts

```diff
@@ -11,27 +12,119 @@ import { RegisterDeviceDto } from '@/routes/notifications/v1/entities/register-d
 import { NotificationsService } from '@/routes/notifications/v1/notifications.service';
 import { ValidationPipe } from '@/validation/pipes/validation.pipe';
 import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import type {
+  UpsertSubscriptionsDto,
+  UpsertSubscriptionsSafesDto,
+} from '@/routes/notifications/v2/entities/upsert-subscriptions.dto.entity';
+import { AuthPayload } from '@/domain/auth/entities/auth-payload.entity';
+import { NotificationType } from '@/domain/notifications/v2/entities/notification.entity';
+import type { UUID } from 'crypto';
+import { NotificationsServiceV2 } from '@/routes/notifications/v2/notifications.service';
+import { recoverMessageAddress } from 'viem';
 
 @ApiTags('notifications')
 @Controller({ path: '', version: '1' })
 export class NotificationsController {
-  constructor(private readonly notificationsService: NotificationsService) {}
+  constructor(
+    // Adding NotificationServiceV2 to ensure compatibility with V1.
+    // @TODO Remove NotificationModuleV2 after all clients have migrated and compatibility is no longer needed.
+    @Inject(NotificationsServiceV2)
+    private readonly notificationServiceV2: NotificationsServiceV2,
+    private readonly notificationsService: NotificationsService,
+  ) {}
 
   @ApiOkResponse()
   @Post('register/notifications')
   @HttpCode(200)
   async registerDevice(
     @Body() registerDeviceDto: RegisterDeviceDto,
   ): Promise<void> {
-    return this.notificationsService.registerDevice(registerDeviceDto);
+    await this.notificationsService.registerDevice(registerDeviceDto);
+
+    // Compatibility with V2
+    // @TODO Remove NotificationModuleV2 after all clients have migrated and compatibility is no longer needed.
+    const compatibleV2Requests =
+      await this.createV2RegisterDto(registerDeviceDto);
+
+    for (const compatibleV2Request of compatibleV2Requests) {
+      await this.notificationServiceV2.upsertSubscriptions(compatibleV2Request);
+    }
```

Comment:
We could use `Promise.all` here.

Reply:
Thanks for pointing this out. I am going to change it but in this case I highly doubt it would result in a noticeable performance change.

Reply:
Done
https://github.com/safe-global/safe-client-gateway/pull/2063/commits/18bd37dfc2ad178017164c69f42ca762d25c2a9f